src/System.Net.Primitives/src/System.Net.Primitives.Friends.cs

```diff
@@ -0,0 +1,8 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+// Cross framework serialization needs access to internals
+[assembly: InternalsVisibleTo("System, PublicKey=00000000000000000400000000000000", AllInternalsVisible=false)]
```

Comment:
Thinking about this a little more we could also just make PathList public in the implementation. @ViktorHofer did you try that?